File: cpp/1257.cpp
#include <iostream>
#include <string>

using namespace std;

int arrayHash(string words[], int elements[], int length) {
    int result = 0, i, j;

    for (i = 0; i < length; i++) {
        for (j = 0; j < words[i].length(); j++) {
            result += (int(words[i][j]) - 65) + j + elements[i];
        }
    }

    return result;
}

int main() {
    int n, i;
    cin >> n;

    for (i = 0; i < n; i++) {
        int l, j, count = 0;
        cin >> l;
        int elements[l];
        string words[l];

        for (j = 0; j < l; j++) {
            cin >> words[j];
            elements[j] = count;

            count += 1;
        }

        cout << arrayHash(words, elements, l) << endl;
    }
}

File: py/1257.py
def arrayHash(words):
    result = 0

    for word, i in words:
        for char in range(len(word)):
            result += (ord(word[char]) - 65) + char + i

    return result


n = int(input())

for _ in range(n):
    words = []
    count = 0
    l = int(input())

    for __ in range(l):
        words.append((input(), count))
        count += 1

    print(arrayHash(words))